src/core/src/op/util/weightless_caching_attributes.cpp

```diff
@@ -4,6 +4,17 @@
 
 #include "openvino/core/rt_info/weightless_caching_attributes.hpp"
 
+#include "openvino/op/util/op_types.hpp"
+
 bool ov::WeightlessCacheAttribute::is_copyable() const {
     return false;
 }
+
+bool ov::WeightlessCacheAttribute::is_copyable(const std::shared_ptr<ov::Node>& from,
+                                               const std::shared_ptr<ov::Node>& to) const {
+    if (!ov::op::util::is_constant(from) || !ov::op::util::is_constant(to)) {
+        return false;
+    }
+
+    return from->get_element_type() != to->get_element_type();
```

Comment:
Can old `is_copy` be used as:
```suggestion
    return !ov::op::util::is_constant(to);
```
For `from` node it shouldn't matter if is constant or not, the attribute is set already. But it will avoid copy it to non-constant node.

For precision check there can be added new matcher pass or transformation which can be GPU specific or common, which will nodes and remove attribute if this node precision is not same as original.
Maybe some existing transformation can do this e.g. ConvertPrecision, @itikhono do you have any suggestion which transformation could be used?

It can be also some helper in GPU when this attribute is get, then check node precision and ignore attribute at some condition.

Reply:
The only way to do it that I can think of is to always return false here and explicitly do the copy during ConvertPrecision pass (and perhaps ConstantFolding). If that's the whole story then it's not a problem. 

However, support for more transformations will follow in a similar manner (https://jira.devtools.intel.com/browse/CVS-157082). The full list is not decided yet but will at least contain layout changes. So we'd have to manually find all transformations which apply data movement ops to constant nodes, do the copying there and also keep in mind that this would have to be repeated in any new transformations added in the future. I don't think that's a viable approach.

I guess I could move the condition somewhere to src/core/src/rt_info.cpp and do something like
`bool copy = is_copyable(to) || condition_func(from, to);` for constants but that's also not great.

Reply:
It it works, could you please add new tests for ConvertPrecision transformation to check that it copies `WeightlessCacheAttribute `.
https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/tests/utils/convert_precision.cpp

Reply:
I can confirm that the approach you describe works for precision conversion, that's what I did in the early drafts of this PR - is_copyable() always returns false and the attr is manually copied in `ConvertPrecision` and `ConstantFolding`. I agree that would be a safer approach with less potential for unexpected things to happen.

The purpose of the change to current approach was to create a structure that makes it possible to add support for more transformations (see second paragraph of https://github.com/openvinotoolkit/openvino/pull/27742#discussion_r1873587940) because I don't think the previous one is suitable for that. But at this point it looks like it'll be best to revert to the approach you describe and figure out the followup design as a separate thing - that's what I'll do.

Reply:
Ok, thanks!

But why do we need to copy this attribute in ConstantFolding in general case?
it looks like we need to copy it only in case when Convert op const folding happens
all other ops might affect the constant data/values and we can't use them from the original bin file.

Reply:
Yeah, I don't remember specifics, but it wasn't for every ConstantFolding case, just in one conditional path which had to do with precision change.

Reply:
> and actually I tried the current implementation with ConstantFolding and it looks like the attribute won't be copied because we do it in 2 steps
> 
> 1. From inputs (Constants which might be marked with `WeightlessCacheAttribute` to the original op (usually not constant)
>    https://github.com/openvinotoolkit/openvino/blob/859958f4e22e76c98121361ec19ea8ddb7cb8233/src/core/src/pass/constant_folding.cpp#L153
> 2. From the original op (usually not constant)  to the new created constant
>    https://github.com/openvinotoolkit/openvino/blob/859958f4e22e76c98121361ec19ea8ddb7cb8233/src/core/src/pass/constant_folding.cpp#L155
> 
> But we expect that from and to both constants.
> 
> In which cases do we expect that the attribute will be copied during ConstantFolding? Probably only ConvertPrecision is affected?

In that previous version the is_copyable() was always false and I copied it manually with a separate method. But I don't remember between what kind of nodes the copying was happening, I'll have to look into it from scratch.

Reply:
I just wanted to highlight that probably the current impl with copyable(from, to) doesn't work for ConstantFolding, and works only with ConvertPrecision. Anyway we need to find another approach.

Also I didn't check it with other transformations which can have 
`copy_runtime_info({node1, node2, const1, const2 etc}, {node3, node4, const3 etc})`
calls. Probably the attribute maybe copied from const1, const2 to const3, looks risky.